### ISSUE TITLE
Qtractoiir Plugin Scan unstable

### ISSUE THREAD

**Issue by TeamB12** [rncbc/qtractor#443]
OS:Artix Linux - Up to date.
Problem: Plugin Scan sometimes doesnt end successfully:
The plugin scan goes to 100% but then it does not finnish but Qtractor is freezing then.
Also, the Scan times of Qtractor is quite different from run to run.
Also the size of the plugin cache files in the config dirare changing in size though no new plugin were added or removed.
It seemed the buffer latency of jack audio server is somehow influencing the outcome, so I am now scanning with 512 buffersize.

**Comment by rncbc** [comment]
you have unstable plugins for sure.

please, before all else avoid scanning to "(Any)" plugin types; try one type at a time.

each turn, check the blacklist and remove the culprits from scan; or start with a very restricted set of plugin paths to scan, this being the first approach to diagnose which plugin(s) is giving you the snags.

**Comment by TeamB12** [comment]
Thanks for your thoughts, but I didnot ask for help,I am trying to do some bug reports.
Yes, not scanning all at once is a good work around - still, sometimes even scanning one plugin type is crashing Qtractor.
Working around is what I am doing every day...its ok for me, but maybe  fixing issues is more then just working around and trying to avoid this ot that.
Also, I do have maybe 4000 plugins installed, I do not caremuch if 30 are not running in Qtractor.
Of course I am deleting blacklist before trying to scan again. Scanning plugins, when  successfull and without a crash is sometimes leading then to different size of the plugin conf files and so to different amount of plugins being successfully scanned.
Different results when doing the same, is maybe a thing to look at.
All plugins from one dev (the one who is developing Qi Granddad) are not recognized by Qtractor.
Also, same results with Ardour and Carla
Bitwig and Reaper are able to recognize and host these plugins, so it is not an issue of the plugins.
Exactly the half amount of orchestools plugins are not being recognized by Qtractor, but they are recognized by Bitwig and Reaper.
If I would find some reason for all this, I will post it here

**Comment by rncbc** [comment]
just a question, are the most of those plugins of which type? vst2, vst3? are they bridged windows plugins for the sake of it? if yes to the last question please avoid them, keep only the sane ones reachable--yes you probably have to get some special paths for them-- also, to not just clear the blacklist, the ones in there are sure to blame (they crash the scan process, for crying out loud).

**Comment by TeamB12** [comment]
I am not counting the installed plugins - but when I am looking at Carla, which is not being able to recognize all plugins installed (like Qtractor and Ardour) then I can see around 3800 plugins. Maybe 150 Clap plugins that are not recognized by Carla (though I have installed ildaeil). Yes maybe 3700 or 3800 plugins installed now. And these are all native Linux plugins.
Sure, I am not using windows plugin and then complaining about DAW problems...
Ok, I will remove the blacklisted plugins from the plugin directories.
Maybe 540 LADSPA , 1800 LV2, 600 VST and 450 VST3 plugins

**Comment by TeamB12** [comment]
Stable running instrument plugin (U-he) gets blacklisted when scanned on audio track...
Scan vst2 plugins, clear blacklist, scan again results allways in different results
When scanning all plugins, no vst2 gets blacklisted
uhe plugins randomly blacklisted without a reason ( using these often)
debug versions of plugins seem to get blacklisted (no debug installed)

Finally, scanning all plugins on an audio track is not blacklisting plugins anymore, the next plugin scan is extremly slow.
Now, there are three blacklisted  plugins though there were no more blacklisted plugins before

next scan is resulting again in an empty blacklist
any another scan brings now the same result: no more blacklisted plugins
Now scanning vst2 plugins is allways blacklisting one plugin.
Repeating this, will allways blacklist another different plugin.
Only vst2 plugins seem to get randomly blacklisted now..

**Comment by rncbc** [comment]
may you test again with qtractor >= 0.9.91.10, please?

**Comment by TeamB12** [comment]
Of course, I can test again with latest Qtractor, but there is a huge problem:
it hasn't landed yet here. I would wait now some days and then I will test again or I have to install outside my packet manager. If it is okay, I will wait some days til the latest Qtractor will get installed through my daily (maybe hourly) updates. If it is urgent, I can setup another testing machine - then you have to cross fingers that I get it installed manually.Reading my first post, I feel like an idiot, I should not mention the number of my installed plugins - I thought it might be the reason why the scan is the way it is. I was over it allready and normally I am starting up Qtractor every day - also I honestly feel not smart enough to debug Qtractor - I guess I have to build it first with debug flag - then again, installing manually? That is my problem...copying everything manually to the right places and then withthe next update,I have to remove evrything again..but testing the new version is of course no problem

**Comment by TeamB12** [comment]
Qtractor Version 1.0 arrived today through my daily updates. I emediatly started testing.  The blacklist now does not show any VST2 Plugins anymore - Thank you!
The scan process seems now to be always the same before it was was scan slow, scan fast - thank you!
Scan seems reliable now - still testing. Again, thank you.

**Comment by TeamB12** [comment]
Now it seems, like plugins are not getting scanned anymore: installed new plugins, these are not found after scan, nor  put in the blacklist . Many several new plugins, I can rule out, that this is for the first time a batch of new plugins that all do not appear at all.

**Comment by rncbc** [comment]
press Ctrl or Shift+Rescan on the plugin selection dialog, to invalidate the cache and force a new inventory.

**Comment by TeamB12** [comment]
Worked. Thanks. Scan now took a while and new plugins are mostly blacklisted, some can be used now in Qtractor.
Only native Linux plugins, no yabridge.
Yes, new plugins here are mostly VST3 and clap, one.
Many of the new plugins blacklisted now:
/home/lee/.vst3/Bandbreite.vst3/Contents/x86_64-linux/Bandbreite.so
/home/lee/.vst3/BYOD.vst3/Contents/x86_64-linux/BYOD.so
/home/lee/.vst3/FreeGain.vst3/Contents/x86_64-linux/FreeGain.so
/home/lee/.vst3/STR-X.vst3/Contents/x86_64-linux/STR-X.so
/home/lee/.vst3/ZL E Compressor.vst3/Contents/x86_64-linux/ZL E Compressor.so
/usr/lib/vst3/ToneLib-TubeWarmth.vst3/Contents/x86_64-linux/ToneLib-TubeWarmth.so
/home/lee/.vst/libOsTIrus.so

Still 3500 plugins usable, I can not see any pattertn which plugins are not running in Qtractor. Sometimes, clap version of a new plugin is not working, but the VST3 and VST2 are ok then.

**Comment by rncbc** [comment]
things might be soewhat better after 7141256 (v1.0.0.6+)

**Comment by TeamB12** [comment]
Finally I got some Qtractor bugs caught on video.
You can watch em here:

[https://www.youtube.com/watch?v=szswZLuEs-Q](url)

Video is 9 minutes long - I am sorry, but, those bugs and issues are really nasty and random.
Making a video felt strange and that I am the first who reports those bugs feels even stranger.
Just watch the last minute, and then the beginning of the video and you can directly see whats going on.
The issues again:
First notes in several midi clips in Qtractor are suddenly missing.
Sometimes, notes are shifted.

Linux is so powerful because everything is perfectly isolated, like the GUI.
How much work would it be to port Qtractor to GTK3 ?

Right now, as video is proofing,I can not work with Qtractor.
Fixing all those strange missing notes and wrong notes allways takes maybe 15 minutes, and then, some minutes later all thoses wrong notes are back again.
Long awaited version 1 seems one of the less reliable Qtractor versions.
